kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/InputStreamPumper.java

```diff
@@ -87,12 +88,14 @@ public static void transferTo(InputStream in, Writable out) throws IOException {
     return CompletableFuture.runAsync(() -> {
       try {
         InputStreamPumper.transferTo(in, out);
-      } catch (Exception e) {
+      } catch (InterruptedIOException e){
         if (!Thread.currentThread().isInterrupted()) {
```

Comment:
We don't really care about this distinction here.  It should just be:

```
      } catch (InterruptedIOException e){
        LOGGER.debug("Interrupted while pumping stream.");
      } catch (Exception e) {
        if (!Thread.currentThread().isInterrupted()) {
          LOGGER.error("Error while pumping stream.", e);
        } else {
          LOGGER.debug("Interrupted while pumping stream.");
        }
      }
```
That is assuming there is something happening where an InterruptedException is then causing an unchecked exception or a non- InterruptedIOException.